### ISSUE TITLE
Accessibility: Dropdown role is combobox which is incorrect

### ISSUE THREAD

**Issue by rickyp-ms** [OfficeDev/office-ui-fabric-react#2335]
### Bug Report

- __Package version(s)__: 4.21.3
- __Browser and OS versions__: All

#### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? Yes

Requested priority: Normal

Products/sites affected: tasks.office.com

#### Describe the issue:

Accessibility review opened a bug on us...
"For the Bucket/Progress Combo box, Voice Over announces "Type text or, to display a list of choices , press Ctrl+option+Space. The part "Type text" is improper and cannot be processed."

Our implementation of the Bucket dropdown looks like this:
```
    <Dropdown
        className="taskBucket"
        label={ TaskDetailsEditorResx.TaskBucketLabel }
        options={ this.getBucketList(task) }
        onChanged={ this.onBucketChanged.bind(this) }
        />
```
Digging through Fabric I did notice the `role="combobox"` on dropdown. My guess is it may supposed to be `role="menu"` so that screen readers do not say `Type text`.

Another choice is if the user is focused on the component and starts typing the name of a list item, we should select that list item. I think the select in browsers do this by default.

#### Actual behavior:

Screen readers read out incorrect information

#### Expected behavior:

Depends if we want to support typing to select an item or just change the role.
